wdl/src/main/scala/wdl/WdlCall.scala

```diff
@@ -82,17 +83,21 @@ object WdlCall {
       *   input3 = other_task.out + 2   -> ExpressionNode with an input port pointing to the output port of other_task.out
       * }
      */
-    def expressionNodeMappings: ErrorOr[Map[LocalName, AnonymousExpressionNode]] = wdlCall.inputMappings traverse {
-      case (inputName, wdlExpression) =>
-        val identifier = wdlCall.womIdentifier.combine(inputName)
-        val constructor = wdlCall match {
-          case _: WdlTaskCall => TaskCallInputExpressionNode.apply _
-          case _ => PlainAnonymousExpressionNode.apply _
-        }
-
-        WdlWomExpression.toAnonymousExpressionNode(identifier, WdlWomExpression(wdlExpression, wdlCall), localLookup, outerLookup, preserveIndexForOuterLookups, wdlCall, constructor) map {
-          LocalName(inputName) -> _
-        }
+    def expressionNodeMappings: ErrorOr[Map[LocalName, AnonymousExpressionNode]] = {
+      val precomputedOgins: Map[String, OutputPort] = outerLookup.filterNot { ol => localLookup.contains(ol._1) } map { case (name, port) => name -> OuterGraphInputNode(WomIdentifier(name), port, preserveIndexForOuterLookups).singleOutputPort }
+      val newLocalLookup = localLookup ++ precomputedOgins
```

Comment:
These are the only new lines (and using `newLocalLookup` below). The rest is reformatting to fit inside the `def`'s new `{` curlies `}`